### PR TITLE
[SYCL] Fix not found kernel due to empty kernel name when using set_arg(s)

### PR DESCRIPTION
**Pull request by alexanderfle**
Using set_arg(s) function in your code with the cacheable kernel leads to the runtime throws an exception "No kernel named  was found -46 (CL_INVALID_KERNEL_NAME)" when the kernel is in process of submitting because the kernel cannot be found in the cache. This patch fixes the issue.

Signed-off-by: Alexander Flegontov <alexander.flegontov@intel.com>

### CODE REVIEW

**Comment by bader** [comment]
@alexanderfle, please, add a regression test.

**Comment by alexanderfle** [comment]
@bader, we already have a regression test for that case, one conformance test that checks handler API.

**Comment by bader** [comment]
> @bader, we already have a regression test for that case, one conformance test that checks handler API.

Interesting... I don't see this test failures in pre-commit or post-commit tests. I think having a small test covering this case in CI will be useful.

File: sycl/test/basic_tests/handler/handler_set_args.cpp [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#include <CL/sycl.hpp>
+#include <cassert>
+
+struct use_offset {
+  static const int no = 0;
+  static const int yes = 1;
+};
```
Comment (alexbatashev) [inline sycl/test/basic_tests/handler/handler_set_args.cpp:18]: Unless you're using fuzzy logic, I believe bool is a more suitable type here.

```suggestion
constexpr bool UseOffset = true;
constexpr bool NoOffset = false;
```

File: sycl/test/basic_tests/handler/handler_set_args.cpp [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+struct single_task_new_functor {
+  single_task_new_functor(accessor_t acc) : acc(acc) {}
+
+  void operator()() { acc[0] = 10; }
+
+  accessor_t acc;
+};
+
+template <int useOffset> struct parallel_for_range_id_functor {
```
Comment (alexbatashev) [inline sycl/test/basic_tests/handler/handler_set_args.cpp:40]:
```suggestion
template <bool useOffset> struct parallel_for_range_id_functor {
```

File: sycl/test/basic_tests/handler/handler_set_args.cpp [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+struct use_offset {
+  static const int no = 0;
+  static const int yes = 1;
+};
+
+using accessor_t =
+    cl::sycl::accessor<int, 1, cl::sycl::access::mode::read_write,
+                       cl::sycl::access::target::global_buffer>;
+
+struct single_task_functor {
```
Comment (alexbatashev) [inline sycl/test/basic_tests/handler/handler_set_args.cpp:24]:
```suggestion
struct SingleTaskFunctor {
```

File: sycl/test/basic_tests/handler/handler_set_args.cpp [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+  single_task_functor(accessor_t acc) : acc(acc) {}
+
+  void operator()() { acc[0] = 10; }
+
+  accessor_t acc;
```
Comment (alexbatashev) [inline sycl/test/basic_tests/handler/handler_set_args.cpp:29]:
```suggestion
  AccessorT MAcc;
```

File: sycl/test/basic_tests/handler/handler_set_args.cpp (hunk 1 above)
Comment (alexanderfle) [reply sycl/test/basic_tests/handler/handler_set_args.cpp]: ok

File: sycl/test/basic_tests/handler/handler_set_args.cpp (hunk 2 above)
Comment (alexanderfle) [reply sycl/test/basic_tests/handler/handler_set_args.cpp]: thanks

File: sycl/test/basic_tests/handler/handler_set_args.cpp (hunk 3 above)
Comment (alexanderfle) [reply sycl/test/basic_tests/handler/handler_set_args.cpp]: done

File: sycl/test/basic_tests/handler/handler_set_args.cpp (hunk 4 above)
Comment (alexanderfle) [reply sycl/test/basic_tests/handler/handler_set_args.cpp]: done

**Comment by alexanderfle** [comment]
@alexbatashev, I have addressed your comments, please review again.

**Comment by alexanderfle** [comment]
@bader, ping

**Pull request merged**